code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -750,6 +750,111 @@
 	var/obj/item/pushbroom/cyborg/BR = locate() in R.model.modules
 	if (BR)
 		R.model.remove_module(BR, TRUE)
+		
+/obj/item/borg/upgrade/condimentsynthesizer
+	name = "Service Cyborg Condiment Synthesiser"
+	desc = "An upgrade to the service model cyborg, allowing it \
+		to produce solid condiments."
+	icon_state = "cyborg_upgrade3"
+	require_model = TRUE
+	model_type = list(/obj/item/robot_model/service)
+	model_flags = BORG_MODEL_SERVICE
+
+/obj/item/borg/upgrade/condimentsynthesizer/action(mob/living/silicon/robot/R, user = usr)
```

Comment:
Should have a better variable name than `R` (even though everywhere else uses `R`, we still want to move away from it)